src/traversal/writeRelayUpdatePayload.js

```diff
@@ -187,6 +199,17 @@ function handleMerge(
     if (!Object.prototype.hasOwnProperty.call(payload, fieldName)) {
       continue;
     }
+
+    // if fieldName is part of deletedIDFieldName in some NODE_DELETE config
+    if (configs.some((config) => {
+      return config.type === RelayMutationType.NODE_DELETE &&
+        Array.isArray(config.deletedIDFieldName) &&
+        config.deletedIDFieldName.length > 0 &&
```

Comment:
Many people would think so. But many others would think the opposite. In other words, it is a bikeshed topic.

One thing that pretty much everyone agrees on, however, is that there is much value in being consistent, picking a pattern and sticking to it, and the pattern we've tried to follow is the one I described.